consent-ws/src/main/java/org/broadinstitute/consent/http/service/DataSetFileParser.java

```diff
@@ -68,25 +73,54 @@ public ParseResult parseTSVFile(File file, List<Dictionary> allFields) {
                 }
             }
         }
-        if ( !errors.isEmpty() ) {
+        if (!errors.isEmpty()) {
             errors.add(PLEASE_DOWNLOAD);
         }
         return errors;
     }
 
-    private List<String> validateRequiredFields(int row, String[] record, List<Dictionary> requiredFields, String id){
+    private List<String> validateRequiredFields(int row, String[] record, List<Dictionary> requiredFields, String id) {
         return requiredFields.stream().filter(field -> record[field.getReceiveOrder()].isEmpty()).map(field -> String.format(BLANK_REQUIRED_FIELD, id, field.getKey(), row)).collect(Collectors.toList());
     }
 
+    private List<String> validateConsentAndCollectionId(String[] record, String datasetName) {
+        if (StringUtils.isEmpty(record[SAMPLE_COLLECTION_INDEX]) && StringUtils.isEmpty(record[CONSENT_ID_INDEX])) {
+            return Arrays.asList(String.format(BLANK_REQUIRED_FIELDS, datasetName));
+        }
+        return Arrays.asList();
+    }
+
     private DataSet createDataSet(String[] record) {
         DataSet dataset = new DataSet();
         dataset.setCreateDate(new Date());
         dataset.setName(record[0]);
-        dataset.setObjectId(record[9]);
+        dataset.setObjectId(StringUtils.isNotEmpty(record[9]) ? record[9] : null);
         dataset.setActive(true);
+        dataset.setConsentName(StringUtils.isNotEmpty(record[10]) ? record[10] : null);
         return dataset;
     }
 
+    public List<DataSet> createAlias(final List<DataSet> dataSets, Integer lastAlias, List<String> predefinedDatasets) {
+        int initialAlias = 3;
+        List<DataSet> results = new ArrayList<>(dataSets);
+        for(DataSet ds: results) {
+            if(StringUtils.isNotEmpty(ds.getName()) && ds.getName().equals(predefinedDatasets.get(1))) {
+                ds.setAlias(2);
+            }
+            else if(StringUtils.isNotEmpty(ds.getName()) && ds.getName().equals(predefinedDatasets.get(0))){
+                ds.setAlias(1);
+            }
+            else if(lastAlias == null || lastAlias < 3) {
+                ds.setAlias(initialAlias);
+                ++initialAlias;
+            }
+            else if(ds.getAlias() == null || ds.getAlias() == 0){
+                ds.setAlias(++lastAlias);
+            }
+        }
```

Comment:
is this code thread safe?

Reply:
My first guess without evidence is that it isn't - there is a single final instance on the service class which is reused for all API calls. That's an existing issue and should be storied out as tech-debt.

Reply:
Here's the story to handle this bug.
https://broadinstitute.atlassian.net/browse/BTRX-407